Sources/NIOHTTP2/DOSHeuristics.swift

```diff
@@ -28,12 +28,15 @@ struct DOSHeuristics<DeadlineClock: NIODeadlineClock> {
     /// The maximum number of "empty" data frames we're willing to tolerate.
     private let maximumSequentialEmptyDataFrames: Int
 
-    private var resetFrameRateControlStateMachine: HTTP2ResetFrameRateControlStateMachine
+    private var resetFrameRateControlStateMachine: RateLimitStateMachine
+    private var streamErrorRateControlStateMachine: RateLimitStateMachine
```

Comment:
Because the state machine reserves capacity on `init` we end up regressing allocations per-connection here; given it's not per-stream I think that's okay.